spec/presenters/user_analytics_presenter_spec.rb

```diff
@@ -1,241 +1,663 @@
 require 'rails_helper'
 
 RSpec.describe UserAnalyticsPresenter, type: :model do
-  let!(:current_user) { create(:user) }
-  let!(:current_facility) { create(:facility, facility_group: current_user.facility.facility_group) }
+  let(:current_user) { create(:user) }
+  let(:request_date) { Date.new(2018, 1, 1) }
 
   describe '#statistics' do
-    context 'daily' do
-      let(:request_date) { Date.new(2018, 1, 1) }
-      let(:reg_date) { request_date - 3.days }
-      let(:follow_up_date) { request_date - 2.days }
+    context 'when diabetes management is enabled' do
+      let(:current_facility) { create(:facility,
+                                      facility_group: current_user.facility.facility_group,
+                                      enable_diabetes_management: true) }
+
+      context 'monthly' do
+        let(:reg_date) { request_date - 2.months }
+        let(:follow_up_date) { request_date - 1.month }
+        let(:controlled_follow_up_date) { request_date }
+
+        before do
+          patients = [create(:patient,
+                             :hypertension,
+                             registration_facility: current_facility,
+                             recorded_at: reg_date,
+                             gender: 'female'),
+                      create(:patient,
+                             :hypertension,
+                             registration_facility: current_facility,
+                             recorded_at: reg_date,
+                             gender: 'male'),
+                      create(:patient,
+                             :diabetes,
+                             registration_facility: current_facility,
+                             recorded_at: reg_date,
+                             gender: 'transgender')]
+
+          patients.each do |patient|
+            create(:encounter,
+                   :with_observables,
+                   observable: create(:blood_pressure,
+                                      :critical,
+                                      patient: patient,
+                                      facility: current_facility,
+                                      user: current_user,
+                                      recorded_at: follow_up_date))
+
+            create(:encounter,
+                   :with_observables,
+                   observable: create(:blood_sugar,
+                                      patient: patient,
+                                      facility: current_facility,
+                                      user: current_user,
+                                      recorded_at: follow_up_date))
+
+            create(:encounter,
```

Comment:
More factories where we should really let the blood_pressure drive, and let encounter be hidden behind the scene.

Reply:
I definitely want to do this, I punted on it since that factory is a little tricky to create.

I'll give it a shot and keep us posted here.